### PR TITLE
Add Random Advice API Option

### PR DESCRIPTION
**Pull request by ndom91**
Added https://api.adviceslip.com/ Option to the Compliments Module.

To use: pass `advice: true` as a config parameter and leave the remoteFile parameter empty. 

Will return a random piece of advice from their API.

### CODE REVIEW

File: modules/default/compliments/compliments.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
+/* global Log, Module, moment */
+
 /* Magic Mirror
  * Module: Compliments
  *
- * By Michael Teeuw https://michaelteeuw.nl
+ * By Michael Teeuw http://michaelteeuw.nl
```
Comment (rejas) [inline modules/default/compliments/compliments.js:6]: why did you undo this?

File: modules/default/compliments/compliments.js [hunk 2]
```diff
@@ -62,7 +65,19 @@ Module.register("compliments", {
 				self.config.compliments = JSON.parse(response);
 				self.updateDom();
 			});
-		}
+		} else if (this.config.advice) {
+      var xobj = new XMLHttpRequest(),
```
Comment (rejas) [inline modules/default/compliments/compliments.js:69]: this needs to be a ; at the end not a ,
thats probably why the tests fail

File: modules/default/compliments/compliments.js [hunk 3]
```diff
@@ -125,7 +140,7 @@ Module.register("compliments", {
 
 		compliments.push.apply(compliments, this.config.compliments.anytime);
 
-		for (var entry in this.config.compliments) {
+		for (entry in this.config.compliments) {
```
Comment (rejas) [inline modules/default/compliments/compliments.js:143]: please do not undo this

File: modules/default/compliments/compliments.js [hunk 4]
```diff
@@ -186,7 +201,7 @@ Module.register("compliments", {
 		// create a span to hold it all
 		var compliment = document.createElement("span");
 		// process all the parts of the compliment text
-		for (var part of parts){
```
Comment (rejas) [inline modules/default/compliments/compliments.js:189]: please do not undo this

File: modules/default/compliments/compliments.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
+/* global Log, Module, moment */
```
Comment (rejas) [inline modules/default/compliments/compliments.js:1]: This isnt neccessary and was already removed

File: modules/default/compliments/compliments.js [hunk 6]
```diff
@@ -39,7 +41,8 @@ Module.register("compliments", {
 		afternoonStartTime: 12,
 		afternoonEndTime: 17,
 		random: true,
-		mockDate: null
+		mockDate: null,
+    advice: false
```
Comment (rejas) [inline modules/default/compliments/compliments.js:45]: please use the same intendation here as in the lines above. makes the code more readable.

File: modules/default/compliments/compliments.js (hunk 2 above)
Comment (rejas) [reply modules/default/compliments/compliments.js:69]: also: please use more intendation here

**Comment by ndom91** [comment]
I'm going to close this one and open another one from a correct branch ;)

**Pull request closed without merging**